src/core/net/srp_server.cpp

```diff
@@ -507,18 +507,25 @@ bool Server::HasNameConflictsWith(Host &aHost) const
         ExitNow(hasConflicts = true);
     }
 
-    for (const Service &service : aHost.mServices)
+    for (const Host &host : mHosts)
     {
-        // Check on all hosts for a matching service with the same
-        // instance name and if found, verify that it has the same
-        // key.
-
-        for (const Host &host : mHosts)
+        for (const Service &service : aHost.mServices)
         {
-            if (host.HasService(service.GetInstanceName()) && (aHost.mKey != host.mKey))
+            // Check on all hosts for a matching service with the same
+            // instance name and if found, verify that it has the same
+            // key.
+
```

Comment:
This is a smaller optimization. The possible downside is that it makes the code harder to read and follow.

Now that we are adding it, perhaps it's better to optimize even further:
- After the first loop, we can add a check: `if (aHost.mKey == host.mKey) { continue; }`
 - Basically, if `host` and `aHost` have the same key, there is no need to check whether their services match or not, since there is no conflict regardless.
 - This way, we skip the whole inner loop.

Reply:
Yes, it's a smaller optimization. I pushed the initial version as you suggested, but I'm not sure whether there were some special considerations in the existing implementation, so I pushed another one switching to the version of a minor change. I'll change it back.